src/Api/SecretsManager/Models/Request/GetSecretsRequestModel.cs

```diff
@@ -1,9 +1,24 @@
 ﻿using System.ComponentModel.DataAnnotations;
+using System.Diagnostics;
 
 namespace Bit.Api.SecretsManager.Models.Request;
 
-public class GetSecretsRequestModel
+public class GetSecretsRequestModel : IValidatableObject
 {
     [Required]
     public IEnumerable<Guid> Ids { get; set; }
+    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        bool isDistinct = Ids.Distinct().Count() == Ids.Count();
+        if (!isDistinct)
+        {
+            IEnumerable<Guid> duplicateGuids = Ids.GroupBy(x => x)
+                                  .Where(g => g.Count() > 1)
+                                  .SelectMany(g => g);
```

Comment:
We can do something like `.Select(g => g.Key)`, so we won't have to Distinct at the bottom.

Reply:
Awesome suggestion, thank you!!